### PR TITLE
Support Object in visit API

### PR DESCRIPTION
**Pull request by jroesch**
This is part of the work in https://github.com/dmlc/tvm/pull/2889, working on cleaning both up this will have to merge first in order to compile the other PR.

### CODE REVIEW

File: src/tvm/node/node.h [hunk 1]
```diff
@@ -23,6 +23,12 @@ namespace runtime {
 class NDArray;
 }  // namespace runtime
 
+namespace relay {
+namespace vm {
```
Comment (tqchen) [inline src/tvm/node/node.h]: perhaps make it runtime::vm::VMObject?

Comment (jroesch) [reply src/tvm/node/node.h]: I was planning to rename right now, to simply the organization, just pushed a commit.

Comment (tqchen) [reply src/tvm/node/node.h]: is VMObject a reference class(like NDArray) or the container itself? Ideally we want it to be the reference class

Comment (tqchen) [reply src/tvm/node/node.h]: could likely be merged into the namspace above(runtime::NDArray)

Comment (jroesch) [reply src/tvm/node/node.h]: It is a reference class, we can move into runtime if you want

Comment (tqchen) [reply src/tvm/node/node.h]: You can make the decison. Perhaps it makes sense to move to runtime given that there is no other runtime Object

Comment (jroesch) [reply src/tvm/node/node.h]: yeah, not sure if we should keep it isolated or not. Maybe it makes sense if we will reuse for AoT/JiT, etc.

File: src/tvm/node/node.h [hunk 2]
```diff
@@ -41,6 +46,7 @@ class EXPORT AttrVisitor {
   virtual void Visit(const char* key, Type* value) = 0;
   virtual void Visit(const char* key, NodeRef* value) = 0;
   virtual void Visit(const char* key, runtime::NDArray* value) = 0;
+  virtual void Visit(const char* key, runtime::vm::Object* value) = 0;
```
Comment (tqchen) [inline src/tvm/node/node.h]: Let us just do runtime::Object then

Comment (jroesch) [reply src/tvm/node/node.h]: I just decided right before you posted this ha. Just pushed a commit to do so.

**Pull request merged**